### PR TITLE
Pr 650 npm

### PR DESCRIPTION
**Pull request by TomasTomecek**
testing out the new fix

### CODE REVIEW

**Comment by TomasTomecek** [comment]
```
Failed to pull image "docker.io/usercont/sandcastle:stg": rpc error: code = Unknown desc = toomanyrequests: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit
```
@_@

**Comment by martinpitt** [comment]
See https://github.com/cockpit-project/cockpit-podman/pull/649 which *should* work now once the 768MB sandcastle fix is rolled out to production.

File: packit.yaml [hunk 1]
```diff
@@ -9,7 +9,8 @@ actions:
   # https://github.com/packit/sandcastle/pull/92
   # https://medium.com/the-node-js-collection/node-js-memory-management-in-container-environments-7eb8409a74e8
   create-archive:
-  - env NODE_OPTIONS=--max-old-space-size=500 make dist-gzip
+  - env NODE_OPTIONS=--max-old-space-size=750 make all
+  - env NODE_OPTIONS=--max-old-space-size=750 make dist-gzip
```
Comment (martinpitt) [inline packit.yaml]: FTR, 300 seems to be the magic number. Big enough to let the build succeed without "out of heap" errors, small enough for 768 MiB RAM.

**Comment by TomasTomecek** [comment]
sadly 300 is not enough :/

```
<--- Last few GCs --->

[125:0x555c5d90ce70]    60007 ms: Mark-sweep 298.8 (302.4) -> 298.4 (302.6) MB, 485.6 / 0.0 ms  (average mu = 0.063, current mu = 0.006) allocation failure scavenge might not succeed


<--- JS stacktrace --->

==== JS stack trace =========================================

    0: ExitFrame [pc: 0x7f3a8e183319]
Security context: 0x0e371b0db989 <JSObject>
    1: /* anonymous */(aka /* anonymous */) [0x14c12a0bffb1] [/sandcastle/docker-io-usercont-sandcastle-stg-20210118-200123416759/node_modules/webpack-sources/lib/applySourceMap.js:~58] [pc=0x1fba6c302ba](this=0x081f6b7804b1 <undefined>,0x13af74be4e39 <String[3]: LT:>,0x2126f2794e51 <Object map = 0x24aaf94ab4a9>)
    2: SourceNode_walk [0x903b9df3981] [/sand...

FATAL ERROR: Ineffective mark-compacts near heap limit Allocation failed - JavaScript heap out of memory
 1: 0x7f3a8d2f007c node::Abort() [/lib64/libnode.so.72]
 2: 0x7f3a8d127588 node::OnFatalError(char const*, char const*) [/lib64/libnode.so.72]
 3: 0x7f3a8d66b78a v8::Utils::ReportOOMFailure(v8::internal::Isolate*, char const*, bool) [/lib64/libnode.so.72]
 4: 0x7f3a8d66ba12 v8::internal::V8::FatalProcessOutOfMemory(v8::internal::Isolate*, char const*, bool) [/lib64/libnode.so.72]
 5: 0x7f3a8d7ff8a9  [/lib64/libnode.so.72]
 6: 0x7f3a8d814267 v8::internal::Heap::PerformGarbageCollection(v8::internal::GarbageCollector, v8::GCCallbackFlags) [/lib64/libnode.so.72]
 7: 0x7f3a8d814fca v8::internal::Heap::CollectGarbage(v8::internal::AllocationSpace, v8::internal::GarbageCollectionReason, v8::GCCallbackFlags) [/lib64/libnode.so.72]
 8: 0x7f3a8d8172ac v8::internal::Heap::AllocateRawWithLightRetry(int, v8::internal::AllocationType, v8::internal::AllocationOrigin, v8::internal::AllocationAlignment) [/lib64/libnode.so.72]
 9: 0x7f3a8d817318 v8::internal::Heap::AllocateRawWithRetryOrFail(int, v8::internal::AllocationType, v8::internal::AllocationOrigin, v8::internal::AllocationAlignment) [/lib64/libnode.so.72]
10: 0x7f3a8d7decaf v8::internal::Factory::NewFillerObject(int, bool, v8::internal::AllocationType, v8::internal::AllocationOrigin) [/lib64/libnode.so.72]
11: 0x7f3a8daee08e v8::internal::Runtime_AllocateInYoungGeneration(int, unsigned long*, v8::internal::Isolate*) [/lib64/libnode.so.72]
12: 0x7f3a8e183319  [/lib64/libnode.so.72]
make: *** [Makefile:75: dist/manifest.json] Aborted (core dumped)
```

**Comment by TomasTomecek** [comment]
500 worked, tinkering with the archive path now

**Comment by TomasTomecek** [comment]
Pending — Build succeeded.

\o/

\o/

\o/

**Comment by martinpitt** [comment]
Awesome, thanks @TomasTomecek ! I submitted that as https://github.com/cockpit-project/cockpit-podman/pull/656 in a slightly cleaner form, without the tests yet (just fixing create-archive). I can't wait to see that land in production :tada: 

BTW, I ran it in production again, and https://prod.packit.dev/srpm-build/12296/logs timed out waiting for the pod to start. That seems to be something entirely unrelated now?

**Comment by martinpitt** [comment]
@TomasTomecek oops! https://prod.packit.dev/srpm-build/12298/logs  says:

persistentvolumeclaims \"sandcastle--sandcastle-20210119-063057534432-pvc\" is forbidden: exceeded quota: 74746f6d6563656b407265646861742e636f6d-noncompute, requested: requests.storage=2Gi, used: requests.storage=43Gi, limited: requests.storage=44Gi","reason":"Forbidden","details":{"name":"sandcastle--sandcastle-20210119-063057534432-pvc","kind":"persistentvolumeclaims"},"code":403}

**Comment by TomasTomecek** [comment]
@martinpitt you're so eager! :)

Sadly our cluster ran out of quota to pull images from docker hub in the morning, the pods deployed only just now. This caused 2 things: our PVC GC didn't run, hence the exceeded quota error, and the worker pods were not updated. All should be live now, gonna check it out.

**Pull request closed without merging**